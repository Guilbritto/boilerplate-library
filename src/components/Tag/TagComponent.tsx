import React from 'react'
import { useTheme } from '../../hooks/useTheme'

import { Container } from './TagComponent.styles'
import { TagProps } from './TagComponent.types'

const TagComponent = ({ label, type }: TagProps) => {
    const theme = useTheme()

    return (
        <Container theme={theme} type={type || 'success'}>
            {label}
        </Container>
    )
}

export default TagComponent
