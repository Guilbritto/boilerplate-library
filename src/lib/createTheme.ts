import { defaultTheme } from "../styles/defaultTheme";

export const createTheme = (theme: typeof defaultTheme) => {
  return {
    ...defaultTheme,
    colors: {
      ...defaultTheme.colors,
      ...theme.colors,
    },
    fontSizes: {
      ...defaultTheme.fontSizes,
      ...theme.fontSizes,
    },
  } as typeof defaultTheme;
}