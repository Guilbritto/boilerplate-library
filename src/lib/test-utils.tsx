import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { ThemeProvider } from 'styled-components'
import { defaultTheme } from '../styles/defaultTheme'

const Providers: FC = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
