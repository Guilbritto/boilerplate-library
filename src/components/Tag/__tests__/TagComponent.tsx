import { render } from '../../../lib/test-utils'

import TagComponent from '../TagComponent'

describe('<TagComponent />', () => {
    it('should render the component', () => {
        const { container } = render(<TagComponent label="teste" />)

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render the success version', () => {
        const { container } = render(
            <TagComponent label="teste" type="success" />
        )

        expect(container.firstChild).toMatchSnapshot()
    })
})
