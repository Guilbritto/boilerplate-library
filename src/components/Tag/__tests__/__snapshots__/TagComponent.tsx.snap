// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagComponent /> should render the component 1`] = `
<div
  class="sc-bdfBQB njdSo"
  type="success"
>
  teste
</div>
`;
